### PR TITLE
Custom menu by user role

### PR DESCRIPTION
**Pull request by danielorkae**
i added custom menu support based on the user role.
for that, i created a helper that checks if the menu exists, so i put the condition that if there is a menu with the authenticated user's role name, then use this menu.

helper name: `menu_exists`

### CODE REVIEW

**Comment by MrCrayon** [comment]
@danielorkae I'm not against it, for what it matter, but I think that whoever decides or not to merge this pull request will be able to take a better decision if you could provide a case scenario in which having a different menu per role could be a better solution than using permissions to have different menu items showing per role as it was suggested many times in the past.

Also in case it gets merged I think it should be made clear in documentation that menu is based on "main" role, since a user can have more than one role.

**Comment by fletch3555** [comment]
I would also like to hear the answer to @MrCrayon's question.  What's the benefit to this over just assigning permissions to your roles?

**Comment by danielorkae** [comment]
I made this PR based on this issue: https://github.com/the-control-group/voyager/issues/4086

I've also had cases where the permissions-only menu did not work for what I'd like to do.
for example, I've created an event registration management website. in it, the inscriptions were grouped in the read of the events resource, and it was necessary that the permission for navigation was activated. so the "inscriptions" item appeared in the menu, but it should not. my solution was to create a specific menu for the "organizer". implemented as in PR.

as this is a solution that depends on whether there is a menu with the same name as the user role, I believe that the two menu modes may coexist in the voyager.
is something I already implement as a rule in my systems, and would be happy to see in the official version.

File: resources/views/dashboard/sidebar.blade.php [hunk 1]
```diff
@@ -30,7 +30,7 @@
 
         </div>
         <div id="adminmenu">
-            <admin-menu :items="{{ menu('admin', '_json') }}"></admin-menu>
+            <admin-menu :items="{{ menu(menu_exists(Auth::user()->role) ? Auth::user()->role : 'admin', '_json') }}"></admin-menu>
```
Comment (fara0) [inline resources/views/dashboard/sidebar.blade.php]: Auth::user()->role->name

**Comment by vkiranmaniya** [comment]
Custom menu  According to the different role is required in some use cases like, Administrator has also the capability to add more Admin/Staff/Management Person with Specific Rules and Authorities. I'm developing this with Voyager and I would like to hear you about it.

**Comment by fletch3555** [comment]
Not planning to add this right now.  You can override the sidebar view and call whatever menu you'd like if needed.

**Comment by neapolis79** [comment]
> Not planning to add this right now. You can override the sidebar view and call whatever menu you'd like if needed.

Excuse me if I resume this closed thread, but personally I think that the ability to manage permissions for different menus for each role is useful and makes sense. Managing admin menu permissions makes sense only if you consider the sidebar of the admin area, but if for example for any site I create a navigation menu to use in the header (as shown in the menu tutorial video) and I have more than one role, I can't manage permissions for this new menu, for any role, not even for the admin one.
The way things are now, I can create more than one menu, but I can completely manage only one. IMHO

**Comment by fletch3555** [comment]
Of course you can. You'll likely have a custom rendering for your menu, so just add the auth checks in there.  Voyager already lets you create/manage permissions, and assign them to roles... so the only part that's missing is the check for that permission.  Since you're talking about rendering these menus outside of Voyager, that specific bit of functionality is outside the scope of Voyager and well within your ability to implement.

**Comment by neapolis79** [comment]
i appreciate it a lot> Of course you can. You'll likely have a custom rendering for your menu, so just add the auth checks in there. Voyager already lets you create/manage permissions, and assign them to roles... so the only part that's missing is the check for that permission. Since you're talking about rendering these menus outside of Voyager, that specific bit of functionality is outside the scope of Voyager and well within your ability to implement.

Yes i'll do for sure because i need in one of my projects, but only to understand and to fullify my curiosity, if rendering menu outside voyager is out of scope, and inside voyager you can only fully manage the admin menu...i think that i'm missing the use case for creating additional menu managing only order and not permission...

Anyway, thank you for your answer and your effort.

**Pull request closed without merging**